File: backjoon/back-2606.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let [C, N, ...cums] = fs.readFileSync(filePath).toString().trim().split("\r\n");

let graph = [];

for (let i = 0; i <= C; i++) {
  graph[i] = [];
}

cums.map((el) => {
  let [idx, num] = el.split(" ");
  graph[idx].push(num);
  graph[num].push(idx);
});

let visited = new Array(+C + 1).fill(false);

let cnt = 0;
dfs(graph, 1, visited);

function dfs(graph, v, visited) {
  visited[v] = true;
  cnt++;
  for (const key of graph[v]) {
    if (!visited[key]) {
      dfs(graph, key, visited);
    }
  }
}

console.log(cnt - 1);

File: back-python/back-2606.py
from logging.config import valid_ident
import sys
sys.setrecursionlimit(10000)
sys.stdin = open("input_py.txt", "r")

C = int(input())
M = int(input())

def dfs(v):
  visited[v] = True

  for i in graph[v]:
    if not visited[i]:
      dfs(i)
graph = [[] for _ in range(C+1)]
visited = [False] * (C+1)

for i in range(M):
  n,v = map(int, sys.stdin.readline().split())
  graph[n].append(v)
  graph[v].append(n)

dfs(1)
cnt =0
for i in visited:
  if i:
    cnt+=1

print(cnt-1)